### ISSUE TITLE
DB::Exception: Too many marks in file skip_idx_sindex_visitorid.cmrk3, marks expected 3 (bytes size 72)

### ISSUE THREAD

**Issue by jasonbigl** [ClickHouse/ClickHouse#53858]
Hi,

I'm try to use the latest version 23.7.4.5 (official build) and skip index, but after I insert data and try to query via skipe index, I got this error, can you please tell me why?

I get started with clickhouse but stuck at first step here

`error: HttpCode:500 ;  ;Code: 33. DB::Exception: Too many marks in file skp_idx_sindex_visitorid.cmrk3, marks expected 3 (bytes size 72). (CANNOT_READ_ALL_DATA) (version 23.7.4.5 (official build))`

Thank you very much!

**Comment by canhld94** [comment]
@jasonbigl can I ask what's the issue you found?

**Comment by jasonbigl** [comment]
> @jasonbigl can I ask what's the issue you found?

Hi, thank you for reply, its just like the issue described, we use skip index and we remove some data periodically. but we found every time we setup with v23.7.4.5, after a few hours, CH start throw that error when we try to query by skip index.

Since its urgent in our side so we fall back to v22.9

I'm gussing this is related to https://github.com/ClickHouse/ClickHouse/issues/52652, it seems its fixed, but at the time I post this issue, the fix is not merged and released, so I closed this issue. we never have time to test it before, but now we are going to test the issue
